sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -498,21 +503,16 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
       throw new TypeError("The parameter 'onError' must be of type 'function'.");
     }
 
-    this._createMessageSessionIfDoesntExist()
-      .then(async () => {
-        if (!this._messageSession) {
-          return;
-        }
-        if (!this._isClosed) {
-          this._messageSession.receive(onMessage, onError, options);
-        } else {
-          await this._messageSession.close();
-        }
-        return;
-      })
-      .catch((err) => {
-        onError(err);
-      });
+    (async () => {
+      if (!this._isClosed) {
+        this._messageSession.receive(onMessage, onError, options);
+      } else {
+        await this._messageSession.close();
+      }
+      return;
+    })().catch((err) => {
+      onError(err);
+    });
```

Comment:
Can this not be

```js
try {
  if (!this._isClosed) {
     this._messageSession.receive(onMessage, onError, options);
  } else {
       await this._messageSession.close()
  }
} 
catch (err) {
    onError(err)
}
```

Reply:
`_registerMessageHandler` is not async, won't be able to use `await`.

Reply:
Ah! thats why you had previously removed the await and added a `.catch()`.
That would be definitely cleaner. How about..
```js
if (!this._isClosed) {
   try {
       this._messageSession.receive(onMessage, onError, options);
   } catch (err) {
       onError(err);
   }
} else {
      this._messageSession.close().catch (err => onError(err));
}
```

Reply:
Yes, yes. Hmmm.. easier to read.

Reply:
Updated